lib/mongoid/traversable.rb

```diff
@@ -119,7 +123,7 @@ class << self
         # an existing field.
         # This condition also checks if the class has any descendants, because
         # if it doesn't then it doesn't need a discriminator key.
-        return unless !fields.key?(discriminator_key) && !descendants.empty?
+        return if fields.key?(discriminator_key) || descendants.empty?
```

Comment:
💪 Love to see [De Morgan's laws](https://en.wikipedia.org/wiki/De_Morgan%27s_laws) in action :)